tests/Squidex.Domain.Apps.Core.Tests/Model/Contents/DefaultContentWorkflowTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// ==========================================================================
+//  Squidex Headless CMS
+// ==========================================================================
+//  Copyright (c) Squidex UG (haftungsbeschränkt)
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using System.Threading.Tasks;
+using Squidex.Domain.Apps.Core.Contents;
+using Squidex.Domain.Apps.Entities.Contents;
+using Squidex.Domain.Apps.Entities.Schemas.State;
+using Xunit;
+
+namespace Squidex.Domain.Apps.Core.Model.Contents
+{
+    public class DefaultContentWorkflowTests
+    {
+        private static readonly DefaultContentWorkflow Workflow = new DefaultContentWorkflow();
+
+        [Fact]
+        public void Get_initial_status_async_tests()
+        {
+            var entity = new SchemaState();
+
+            var result = Workflow.GetInitialStatusAsync(entity);
+
+            Assert.IsType<Task<Status2>>(result);
+            Assert.Equal("Draft", result.Result.Name);
+        }
+
+        [Fact]
+        public void Is_valid_next_status_tests()
+        {
+            var entity = new ContentEntity();
+
+            var status = new Status2("Draft");
+
+            var result = Workflow.IsValidNextStatus(entity, status);
+
+            Assert.IsType<Task<bool>>(result);
+            Assert.True(result.Result);
+        }
+
+        [Fact]
+        public void Can_update_async_tests()
+        {
+            var entity = new ContentEntity();
+
+            var result = Workflow.CanUpdateAsync(entity);
+
+            Assert.IsType<Task<bool>>(result);
+            Assert.True(result.Result);
+        }
+
+        [Fact]
+        public void Get_nexts_async_tests()
+        {
+            var draftContent = new ContentEntity() { Status = Status.Draft };
```

Comment:
I would not use the ContentEntity class here, because this would create the coupling to another class and in generel ContentEntity is just an implementation detail of ContentQueryService.

I would create mocks for it  and write a helper method:

https://github.com/Squidex/squidex/blob/master/tests/Squidex.Domain.Apps.Entities.Tests/Contents/ContentQueryServiceTests.cs#L547